### ISSUE TITLE
Precondition Failed on Startup

### ISSUE THREAD

**Issue by reststop** [czechboy0/Buildasaur#268]
Receive the following upon restarting Buildasaur from within Xcode
Not sure how repeatable, yet.

Workaround has been to remove the Buildasaur directory from ~/Library/Application Support.

Log shows:

<pre>
Buildasaur 1.0.0-b2 launched at 2016-05-14 00:54:50 +0000.
*
*
*

precondition failed: : file /Users/myself/buildasaur/buildasaur/BuildaGitServer/Base/BaseTypes.swift, line 38
</pre>

<pre>
    public func createServer(service: GitService, auth: ProjectAuthenticator?) -> SourceServerType {
        
        if let auth = auth {
            precondition(service.serviceType() == auth.service.serviceType())
        }
        
        return GitServerFactory.server(service, auth: auth)
    }
</pre>

**Comment by reststop** [comment]
I have not yet checked to see if this is a branch change or a master change.  Just filing an issue for a HEADS-UP since your assistance has been fantastic on pointing me to root causes.

It has happened 2 times.  Both times, the only common thing I know happened, was syncing my local branch to the repository, and then doing a "run" in Xcode.

**Comment by czechboy0** [comment]
Yeah, please see if that's something specific to the fork you're using. Also, if you could provide a sample project that reproduces it that'd be gold 👍

**Comment by reststop** [comment]
I do have both of the offending ~/Library/Application Support/Buildasaur directories available.
In each case, I simply renamed the directory, and let Buildasaur recreate a new correct one.
Each of these was working at one point, then stopped working when I restarted Buildsasur.
I believe that both of the errant configurations were setup with a single syncer for the Buildasaur project.

I've included the 2nd one here.  Be happy to provide the first one as well if it will help.
I have redacted any identifying info from the logs and config, replacing the username with "me" and hostnames with mycompany.

[Buildasaur-two.zip](https://github.com/czechboy0/Buildasaur/files/267044/Buildasaur-two.zip)

I looked through the files and didn't see anything out of the ordinary.  Did a diff against the live directory and other than the UUIDs and usernames I redacted, there were no obvious differences.

**Comment by czechboy0** [comment]
In ServerConfigs.json, I can see `"user" : “me”`, but the second set of quotes are an incorrect type. Did you do that when anonymizing the data or was it there before?
